Remotion/Web/Development.WebTesting/Accessibility/AccessibilityAnalyzer.cs

```diff
@@ -36,10 +36,34 @@ public class AccessibilityAnalyzer
   {
     private static readonly TimeSpan s_defaultMaximumTimeToWaitForFrame = TimeSpan.FromMilliseconds(3000);
 
+    /// <summary>
+    /// Creates an <see cref="AccessibilityAnalyzer"/> that uses an instance of <see cref="IWebDriver"/> for both <see cref="IWebDriver"/>
+    /// and <see cref="IJavaScriptExecutor"/>.
+    /// </summary>
+    [NotNull]
+    public static AccessibilityAnalyzer CreateForWebDriver (
+        [NotNull] IWebDriver webDriver,
+        [NotNull] IAxeResultParser axeResultParser,
+        [NotNull] IAccessibilityConfiguration configuration,
+        [NotNull] IAxeSourceProvider sourceProvider,
+        [NotNull] IAccessibilityResultMapper mapper,
+        [NotNull] ILog logger)
+    {
+      ArgumentUtility.CheckNotNull("webDriver", webDriver);
+      ArgumentUtility.CheckNotNull("axeResultParser", axeResultParser);
+      ArgumentUtility.CheckNotNull("configuration", configuration);
+      ArgumentUtility.CheckNotNull("sourceProvider", sourceProvider);
+      ArgumentUtility.CheckNotNull("mapper", mapper);
+      ArgumentUtility.CheckNotNull("logger", logger);
+
+      return new AccessibilityAnalyzer(webDriver, (IJavaScriptExecutor)webDriver, axeResultParser, configuration, sourceProvider, mapper, logger);
+    }
+
     /// <summary>
     /// Creates an <see cref="AccessibilityAnalyzer"/> that uses an instance of <see cref="RemoteWebDriver"/> for both <see cref="IWebDriver"/>
     /// and <see cref="IJavaScriptExecutor"/>.
     /// </summary>
+    [Obsolete("Use CreateForWebDriver(IWebDriver, IAxeResultParser, IAccessibilityConfiguration, IAxeSourceProvider, IAccessibilityResultMapper, ILog) instead. ", true)]
```

Comment:
```suggestion
    [Obsolete("Use CreateForWebDriver(IWebDriver, IAxeResultParser, IAccessibilityConfiguration, IAxeSourceProvider, IAccessibilityResultMapper, ILog) instead. (Version: 3.20.0)", false)]
```

I always add the version when the obsolete was introduced. It allows for efficient removal of obsolete-attributes that are old enough to no longer serve a purpose during migration. Typically, in the next major version.

The boolea flag specifies if the attribute is a warning or an error. Code that still works as expected should not be fitted with an error-level obsolete-attribute. And if you do add an error-level obsolete, the implementation should be replaced with a NotSupportedException (using the attribute's text as the message). That way, uneeded dependencies can be dropped and in general, no longer invokable code is removed.

Please check for things such as trailing spaces in messages.